proof/tx.go

```diff
@@ -167,3 +181,65 @@ func (p *TxMerkleProof) Decode(r io.Reader) error {
 
 	return nil
 }
+
+// TxProof is a struct that contains all the necessary elements to prove the
+// existence of a certain outpoint in a block.
+type TxProof struct {
+	// MsgTx is the transaction that contains the outpoint.
+	MsgTx wire.MsgTx
+
+	// BlockHeader is the header of the block that contains the transaction.
+	BlockHeader wire.BlockHeader
+
+	// BlockHeight is the height at which the block was mined.
+	BlockHeight uint32
+
+	// MerkleProof is the proof that the transaction is included in the
+	// block and its merkle root.
+	MerkleProof TxMerkleProof
+
+	// ClaimedOutPoint is the outpoint that is being proved to exist in the
+	// transaction.
+	ClaimedOutPoint wire.OutPoint
+}
```

Comment:
Anyone can prove an outpoint in a block. The UTXO set is public. Any sender could use up all UTXOs.

Maybe I'm missing something, please see https://github.com/lightninglabs/taproot-assets/issues/874#issuecomment-2858564454

Reply:
Addressed this by adding the required internal key (and optional merkle root) fields to the TX proof.